api/middleware/responseLogger_test.go

```diff
@@ -0,0 +1,203 @@
+package middleware
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/api/address"
+	"github.com/ElrondNetwork/elrond-go/api/mock"
+	"github.com/ElrondNetwork/elrond-go/api/wrapper"
+	"github.com/ElrondNetwork/elrond-go/config"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/gin-contrib/cors"
+	"github.com/gin-gonic/gin"
+	"github.com/stretchr/testify/assert"
+)
+
+func startNodeServerResponseLogger(handler address.FacadeHandler, respLogMiddleware *responseLoggerMiddleware) *gin.Engine {
+	ws := gin.New()
+	ws.Use(cors.Default())
+	ws.Use(respLogMiddleware.MiddlewareHandlerFunc())
+	ginAddressRoutes := ws.Group("/address")
+	if handler != nil {
+		ginAddressRoutes.Use(WithFacade(handler))
+	}
+	addressRoutes, _ := wrapper.NewRouterWrapper("address", ginAddressRoutes, getRoutesConfig())
+	address.Routes(addressRoutes)
+	return ws
+}
+
+type responseLogFields struct {
+	title    string
+	path     string
+	request  string
+	duration time.Duration
+	status   int
+	response string
+}
+
+func TestNewResponseLoggerMiddleware(t *testing.T) {
+	t.Parallel()
+
+	rlm := NewResponseLoggerMiddleware(10)
+
+	assert.False(t, check.IfNil(rlm))
+}
+
+func TestResponseLoggerMiddleware_DurationExceedsTimeout(t *testing.T) {
+	t.Parallel()
+
+	thresholdDuration := 10 * time.Millisecond
+	addr := "testAddress"
+	facade := mock.Facade{
+		BalanceHandler: func(s string) (i *big.Int, e error) {
+			time.Sleep(thresholdDuration + 1*time.Millisecond)
+			return big.NewInt(37777), nil
+		},
+	}
+
+	rlf := responseLogFields{}
+	printHandler := func(title string, path string, duration time.Duration, status int, request string, response string) {
+		rlf.title = title
+		rlf.path = path
+		rlf.duration = duration
+		rlf.status = status
+		rlf.response = response
+		rlf.request = request
+	}
+
+	rlm := NewResponseLoggerMiddleware(thresholdDuration)
+	rlm.printRequestFunc = printHandler
```

Comment:
can remain as it is. If concurrency errors happen because of this, might consider to create a new constructor in `expiort_test.go` on which you can inject the print handler.

Reply:
for now, tests pass even if I set `-race` mode when running tests